bqskit/runtime/__init__.py

```diff
@@ -162,6 +162,16 @@ def cancel(self, future: RuntimeFuture) -> None:
         """
         ...
 
+    def get_cache(self) -> dict[str, Any]:
+        """
+        Retrieve worker's local cache.
+
+        A worker's local cache is implemented as a dictionary. This function can
+        be used to check the presence of data, or to place data directly into, a
+        worker's local cache.
+        """
```

Comment:
What is a "worker's local cache"? The documentation here should explain how this can be used while abstracting details of how the runtime works. Consider a user reading this may not know (nor should they know) what a worker is. What is important is that this cache is transient. You can place things in here, to reduce reloading, but they may not be in here later. This informs the user that when using this cache, you should write something like:

```python
def load_expensive_io_model_or_something_or_another(file):
    cache = get_runtime().get_cache()
    if some_file_name_key in cache:
        return that
    # load it; store in cache; return it
    ...
```

Not saying you can't or shouldn't put implementation details later in the docstring, but the first points should inform the user how to use this method/function/class/etc. In this case, definitely include somewhere that this cache is local to a specific worker and is never serialized/pickled.

Reply:
Removed reference to workers here. Added example as to how to use the cache.